frontend/src/metabase/collections/containers/CollectionSidebar/Collections/CollectionsList/CollectionsList.unit.spec.js

```diff
@@ -1,72 +1,131 @@
 
+import { PLUGIN_COLLECTIONS } from "metabase/plugins";
+
 import CollectionsList from "./CollectionsList";
 
-const filter = () => true;
+describe("CollectionsList", () => {
+  function setup({ collections = [], openCollections = [], ...props } = {}) {
+    render(
+      <DragDropContextProvider backend={HTML5Backend}>
+        <CollectionsList
+          collections={collections}
+          openCollections={openCollections}
+          filter={() => true}
+          {...props}
+        />
+      </DragDropContextProvider>,
+    );
+  }
+
+  function collection({
+    id,
+    name = "Collection name",
+    authority_level = null,
+    location = "/",
+    children = [],
+    archived = false,
+  } = {}) {
+    return {
+      id,
+      name,
+      authority_level,
+      location,
+      children,
+      archived,
+    };
+  }
+
+  it("renders a basic collection", () => {
+    setup({
+      collections: [collection({ id: 1, name: "Collection name" })],
+    });
+
+    expect(screen.queryByText("Collection name")).toBeVisible();
+  });
 
-const openCollections = [];
+  it("opens child collection when user clicks on chevron button", () => {
+    const onOpen = jest.fn();
+    setup({
+      collections: [
+        collection({
+          id: 1,
+          name: "Parent collection name",
+          children: [
+            collection({
+              id: 2,
+              name: "Child collection name",
+              location: "/2/",
+            }),
+          ],
+        }),
+      ],
+      onOpen,
+    });
 
-it("renders a basic collection", () => {
-  const collections = [
-    {
-      archived: false,
-      children: [],
+    userEvent.click(screen.getByLabelText("chevronright icon"));
+
+    expect(onOpen).toHaveBeenCalled();
+  });
+
+  describe("Collection types", () => {
+    const regularCollection = collection({ id: 1, authority_level: null });
+    const officialCollection = collection({
       id: 1,
-      location: "/",
-      name: "Collection name",
-    },
-  ];
-
-  render(
-    <DragDropContextProvider backend={HTML5Backend}>
-      <CollectionsList
-        collections={collections}
-        filter={filter}
-        openCollections={openCollections}
-      />
-    </DragDropContextProvider>,
-  );
-
-  screen.getByText("Collection name");
-});
+      authority_level: "official",
+    });
+
+    describe("OSS", () => {
+      it("displays folder icon for regular collections", () => {
+        setup({ collections: [regularCollection] });
+        expect(screen.queryByLabelText("folder icon")).toBeInTheDocument();
+        expect(screen.queryByLabelText("badge icon")).toBeNull();
+      });
+
+      it("displays folder icon for official collections", () => {
+        setup({ collections: [officialCollection] });
+        expect(screen.queryByLabelText("folder icon")).toBeInTheDocument();
+        expect(screen.queryByLabelText("badge icon")).toBeNull();
+      });
+    });
+
+    describe("EE", () => {
```

Comment:
nice setup here with OSS/EE. we should definitely try to think of ways to make this easier